### PR TITLE
[PF-2364] Bundle the command docs in release

### PR DESCRIPTION
**Pull request by NavidZ**
Add picocli command doc generation to gradle and
include the doc artifacts in the release distribution
bundle. Also update docs for a few commands.

This change is based on the gj/doc-gen branch. But at this
point it only has the command doc generation stuff. The readme
doc and some additional instructions in the original branch
need further cleanup and breaking into separate articles.

The updated docs appear like this in the console for the updated commands:
```
$ terra
Terra command-line interface.
Usage: terra [COMMAND]
The Terra CLI allows advanced users to interact with Terra workspaces and
resources, and to perform administrative actions.

Example usage:
- Fetch the user's credentials and check the authentication status.
    terra auth login
    terra auth status

- Ping the Terra server.
    terra server status

- Create a new Terra workspace and backing Google project. Check the current
context to confirm it was created successfully.
    terra workspace create
    terra status

- List all workspaces the user has read or write access to.
    terra workspace list

- To use an existing Terra workspace, use the set command instead of create.
    terra workspace set --id=<workspace-id>

- Create a Terra-managed bucket for temporary data storage.
    terra resource create gcs-bucket --name=<name> --bucket-name=<name>

Commands:
  app        Run applications in the workspace.
  auth       Commands to manage user credentials.
  bq         Call bq in the Terra workspace.
  config     Commands to configure the CLI.
  cromwell   Commands related to Cromwell workflows.
  gcloud     Call gcloud in the Terra workspace.
  git        Call git in the Terra workspace.
  group      Manage groups of Terra users.
  gsutil     Call gsutil in the Terra workspace.
  nextflow   Call nextflow in the Terra workspace.
  notebook   Use GCP Notebooks in the workspace.
  resolve    Resolve a resource to its cloud id or path.
  resource   Manage references and controlled resources in the workspace.
  server     Connect to a Terra server.
  spend      Manage spend profiles.
  status     Print details about the current workspace and server.
  user       Manage users.
  version    Get the installed version.
  workspace  Setup a Terra workspace.
Exit codes:
  0         Successful program execution
  1         User-actionable error (e.g. missing parameter, workspace not
             defined in the current context)
  2         System or internal error (e.g. error response from the Terra server)
  3         Unexpected error (e.g. Java exception)
  other     Third-party application exit codes will be passed through to the
             caller. For example, if `gcloud --malformedOption` returns exit
             code `2`, then `terra gcloud --malformedOption` will also return
             exit code `2`.


$ terra app
Missing required subcommand
Run applications in the workspace.
Usage: terra app [COMMAND]
The Terra CLI allows running supported third-party applications within the
context of a workspace. The `app-launch` config property controls how tools are
run: either in a Docker container, or as a child process.

Nextflow and the `gcloud` SDK are the first examples of supported third-party
applications.
Commands:
  execute  [FOR DEBUG] Execute a command in the application container for the
             Terra workspace, with no setup.
  list     List the supported applications.
```

Also after ./gradlew installDist the install package is visible under build directory that has autogenerated doc files:

```
$ ls build/install/
terra-cli
$ ls build/install/terra-cli/
README.md	bin		docs		install.sh	lib
$ ls build/install/terra-cli/docs/
commands
$ ls build/install/terra-cli/docs/commands/
terra-app-execute.adoc			terra-config-set.adoc			terra-resource-check-access.adoc	terra-spend-list-users.adoc
terra-app-list.adoc			terra-config.adoc			terra-resource-create-bq-dataset.adoc	terra-spend.adoc
terra-app.adoc				terra-cromwell-generate-config.adoc	terra-resource-create-gcp-notebook.adoc	terra-status.adoc
terra-auth-login.adoc			terra-cromwell.adoc			terra-resource-create-gcs-bucket.adoc	terra-user-invite.adoc
terra-auth-revoke.adoc			terra-gcloud.adoc			terra-resource-create.adoc		terra-user-ssh-key-generate.adoc
terra-auth-status.adoc			terra-git.adoc				terra-resource-delete.adoc		terra-user-ssh-key-get.adoc
terra-auth.adoc				terra-group-add-user.adoc		terra-resource-describe.adoc		terra-user-ssh-key.adoc
terra-bq.adoc				terra-group-create.adoc			terra-resource-list-tree.adoc		terra-user-status.adoc
terra-config-get-app-launch.adoc	terra-group-delete.adoc			terra-resource-list.adoc		terra-user.adoc
terra-config-get-browser.adoc		terra-group-describe.adoc		terra-resource-resolve.adoc		terra-version.adoc
terra-config-get-format.adoc		terra-group-list-users.adoc		terra-resource-update-bq-dataset.adoc	terra-workspace-add-user.adoc
terra-config-get-image.adoc		terra-group-list.adoc			terra-resource-update-bq-table.adoc	terra-workspace-break-glass.adoc
terra-config-get-logging.adoc		terra-group-remove-user.adoc		terra-resource-update-gcp-notebook.adoc	terra-workspace-clone.adoc
terra-config-get-resource-limit.adoc	terra-group.adoc			terra-resource-update-gcs-bucket.adoc	terra-workspace-create.adoc
terra-config-get-server.adoc		terra-gsutil.adoc			terra-resource-update-gcs-object.adoc	terra-workspace-delete-property.adoc
terra-config-get-workspace.adoc		terra-nextflow.adoc			terra-resource-update-git-repo.adoc	terra-workspace-delete.adoc
terra-config-get.adoc			terra-notebook-start.adoc		terra-resource-update.adoc		terra-workspace-describe.adoc
terra-config-list.adoc			terra-notebook-stop.adoc		terra-resource.adoc			terra-workspace-list-users.adoc
terra-config-set-app-launch.adoc	terra-notebook.adoc			terra-server-list.adoc			terra-workspace-list.adoc
terra-config-set-browser.adoc		terra-resolve.adoc			terra-server-set.adoc			terra-workspace-remove-user.adoc
terra-config-set-format.adoc		terra-resource-add-ref-bq-dataset.adoc	terra-server-status.adoc		terra-workspace-set-property.adoc
terra-config-set-image.adoc		terra-resource-add-ref-bq-table.adoc	terra-server.adoc			terra-workspace-set.adoc
terra-config-set-logging.adoc		terra-resource-add-ref-gcs-bucket.adoc	terra-spend-create-profile.adoc		terra-workspace-update.adoc
terra-config-set-resource-limit.adoc	terra-resource-add-ref-gcs-object.adoc	terra-spend-delete-profile.adoc		terra-workspace.adoc
terra-config-set-server.adoc		terra-resource-add-ref-git-repo.adoc	terra-spend-disable.adoc		terra.adoc
terra-config-set-workspace.adoc		terra-resource-add-ref.adoc		terra-spend-enable.adoc

```

### CODE REVIEW

**Comment by NavidZ** [comment]
@dexamundsen do you happen to know what's going on the with bots and tests in this repo? I noticed on this PR that the tests seem to be quite flaky and I had to rerun them to get one of the CIs passed. The other one is not passing even on an almost empty PR.

File: gradle/tools.gradle [hunk 1]
```diff
@@ -19,3 +19,13 @@ task getDockerImageTag {
         println("${project.properties['dockerImageTag']}")
     }
 }
+
+task generateDocs (type: JavaExec) {
```
Comment (dexamundsen) [inline gradle/tools.gradle:23]: Could we also add verification in the PR that the docs are published and bundled as expected? 
steps - https://github.com/DataBiosphere/terra-cli/blob/mc-feature/CONTRIBUTING.md

**Comment by dexamundsen** [comment]
> @dexamundsen do you happen to know what's going on the with bots and tests in this repo? I noticed on this PR that the tests seem to be quite flaky and I had to rerun them to get one of the CIs passed. The other one is not passing even on an almost empty PR.

These are baseline failure - https://github.com/DataBiosphere/terra-cli/actions/runs/3820427810/jobs/6498732359
I am having a look at these

File: gradle/tools.gradle (hunk 1 above)
Comment (NavidZ) [reply gradle/tools.gradle:23]: I added some notes to the description. But just to confirm what you mean. Are you suggesting to add a Java unit test to verify the content of the distribution folder? as I don't see any similar test in the repo verifying the gradle tasks themselves and whether they are setup correctly.

File: gradle/tools.gradle [hunk 2]
```diff
@@ -19,3 +19,13 @@ task getDockerImageTag {
...
+    dependsOn(classes)
+    group = "Documentation"
+    description = "Generate Terra CLI docs"
+    classpath(configurations.annotationProcessor, sourceSets.main.runtimeClasspath)
+    main 'picocli.codegen.docgen.manpage.ManPageGenerator'
+    args 'bio.terra.cli.command.Main', "--outdir=${project.buildDir}/docs/commands/"
+}
+assemble.dependsOn tasks.generateDocs
```
[the remark this answers is not in the document]
Comment (NavidZ) [reply gradle/tools.gradle]: done

File: src/main/java/bio/terra/cli/command/Main.java [hunk 3]
```diff
@@ -27,6 +27,44 @@
  */
 @Command(
     name = "terra",
+    header = "Terra command-line interface.",
```
[the remark this answers is not in the document]
Comment (NavidZ) [reply src/main/java/bio/terra/cli/command/Main.java:30]: done

File: gradle/tools.gradle (hunk 1 above)
Comment (dexamundsen) [reply gradle/tools.gradle:23]: this works

File: src/main/java/bio/terra/cli/command/Main.java [hunk 4]
```diff
@@ -27,6 +27,44 @@
...
+    description =
+        "The Terra CLI allows advanced users to interact with Terra workspaces and resources, and to perform administrative actions. \n\n"
+            + "== Example usage \n\n"
```
Comment (dexamundsen) [inline src/main/java/bio/terra/cli/command/Main.java:33]: minor: could reduce the blank spaces and newlines to keep this compact

File: src/main/java/bio/terra/cli/command/Main.java [hunk 5]
```diff
@@ -27,6 +27,44 @@
...
+            + "terra workspace create \n\n"
+            + "terra status \n\n"
+            + "....\n\n"
+            + "List all workspaces the user has read or write access to. \n\n"
+            + "....\n\n"
+            + "terra workspace list \n\n"
+            + "....\n\n"
+            + "To use an existing Terra workspace, use the set command instead of create. \n\n"
+            + "....\n\n"
+            + "terra workspace set --id=eb0753f9-5c45-46b3-b3b4-80b4c7bea248 \n\n"
```
Comment (dexamundsen) [inline src/main/java/bio/terra/cli/command/Main.java:54]: minor: something like --id=[workspace-id] to avoid confusion ?

**Review by dexamundsen — APPROVED** [review]
minor comments, not blockers

File: src/main/java/bio/terra/cli/command/Main.java (hunk 5 above)
Comment (NavidZ) [reply src/main/java/bio/terra/cli/command/Main.java]: done.

**Review by gjuggler — APPROVED** [review]

**Pull request merged**